lib/players/computer.rb

```diff
@@ -0,0 +1,93 @@
+class Player::Computer < Player
+  attr_accessor :token, :combos
+
+  def initialize(token)
+    @token = token
+    @combos = Game::WIN_COMBINATIONS
+    @opponents_token = "O" if @token == "X"
+    @opponents_token = "X" if @token == "O"
+  end
+
+  def move(board)
+
+    move_index =
+    case
+    when take_middle(board)
+      take_middle(board)
+    when win(board)
+      win(board)
+    when block(board)
+      block(board)
+    when corner(board)
+      corner(board)
+    else
+      random(board)
+    end
```

Comment:
One downside to the `case` statement approach here is that you will always execute one of these methods twice (once for the `when` criteria and once for the return value).
Because of the way you've structured your methods, you could also do something like this:
```
move_index = take_middle(board) || win(board) || block(board) || corner(board) || random(board)
```
If you like this approach, I would say that line is looking a little long. Maybe it makes sense to group some of those methods together into another method?